vnet/spec/vnet/models/segment_spec.rb

```diff
@@ -0,0 +1,22 @@
+# -*- coding: utf-8 -*-
+require 'spec_helper'
+
+describe Vnet::Models::Segment do
+  describe "destroy" do
+    it "deletes associated mac_leases and mac_addresses" do
```

Comment:
We also need to add an additional for the cases where the segment is associated with a network (and ip leases).